File: Exerc5/MidHandler/src/message/MessageHeader.java
package message;

import java.io.Serializable;

public class MessageHeader implements Serializable{
	/**
	 * 
	 */
	private static final long serialVersionUID = 9159746341141521538L;
	private String magic;
	private int version;
	private boolean byteOrder;
	private int messageType;
	private long messageSize;
	
	
	
	/**
	 * @param magic
	 * @param version
	 * @param byteOrder
	 * @param messageType
	 * @param messageSize
	 */
	public MessageHeader(String magic, int version, boolean byteOrder, int messageType, long messageSize) {
		super();
		this.magic = magic;
		this.version = version;
		this.byteOrder = byteOrder;
		this.messageType = messageType;
		this.messageSize = messageSize;
	}
	/**
	 * @return the magic
	 */
	public String getMagic() {
		return magic;
	}
	/**
	 * @param magic the magic to set
	 */
	public void setMagic(String magic) {
		this.magic = magic;
	}
	/**
	 * @return the version
	 */
	public int getVersion() {
		return version;
	}
	/**
	 * @param version the version to set
	 */
	public void setVersion(int version) {
		this.version = version;
	}
	/**
	 * @return the byteOrder
	 */
	public boolean isByteOrder() {
		return byteOrder;
	}
	/**
	 * @param byteOrder the byteOrder to set
	 */
	public void setByteOrder(boolean byteOrder) {
		this.byteOrder = byteOrder;
	}
	/**
	 * @return the messageType
	 */
	public int getMessageType() {
		return messageType;
	}
	/**
	 * @param messageType the messageType to set
	 */
	public void setMessageType(int messageType) {
		this.messageType = messageType;
	}
	/**
	 * @return the messageSize
	 */
	public long getMessageSize() {
		return messageSize;
	}
	/**
	 * @param messageSize the messageSize to set
	 */
	public void setMessageSize(long messageSize) {
		this.messageSize = messageSize;
	}
}

File: Projeto/PythonServer/message/MessageHeader.py
import message_pb2

class MessageHeader(object):
    #--------------Constructor--------------
    def __init__(self,magic,version,byteOrder,messageType,messageSize):
        self.__messageHeader = message_pb2.MessageHeader()
        self.__messageHeader.magic = magic
        self.__messageHeader.version = version
        self.__messageHeader.byteOrder = byteOrder
        self.__messageHeader.messageType = messageType
        self.__messageHeader.messageSize = messageSize
    
    #---------------------------------------

    #----------Getters and Setters----------
    def getMagic(self):
        return self.__messageHeader.magic

    def getVersion(self):
        return self.__messageHeader.version

    def getByteOrder(self):
        return self.__messageHeader.byteOrder

    def getMessageType(self):
        return self.__messageHeader.messageType

    def getMessageSize(self):
        return self.__messageHeader.messageSize

    def setMagic(self,magic):
        self.__messageHeader.magic = magic

    def setVersion(self,version):
        self.__messageHeader.version = version

    def setByteOrder(self,byteOrder):
        self.__messageHeader.byteOrder = byteOrder

    def setMessageType(self,messageType):
        self.__messageHeader.messageType = messageType

    def setMessageSize(self,messageSize):
        self.__messageHeader.messageSize = messageSize

    #---------------------------------------